### ISSUE TITLE
tab size is set to 8 in latest version

### ISSUE THREAD

**Issue by volksvagen** [microsoft/vscode#126794]
Issue Type: <b>Performance Issue</b>

as title, default tab size is set to 8 in latest version, which is an unbelievable modification.

VS Code version: Code 1.57.1 (507ce72a4466fbb27b715c3722558bb15afa9f48, 2021-06-17T13:28:07.755Z)
OS version: Windows_NT x64 10.0.19041
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8350U CPU @ 1.70GHz (8 x 1896)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.85GB (7.04GB free)|
|Process Argv|--crash-reporter-id ffd39a9b-3af7-4188-ba19-d2e6963bbef3|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    3	   123	  4488	code main
    0	    19	  1680	   "C:\Program Files (x86)\SogouInput\9.8.0.3746\SGTool.exe" -daemon
    0	   180	  4156	   gpu-process
    1	    82	  6240	   window (Issue Reporter)
    0	   175	  9580	   window (● Untitled-1 - Visual Studio Code)
    0	   144	   504	     extensionHost
    0	    68	 10432	       electron_node serverMain.js 
    0	    70	 12252	       "C:\Users\epuxphi\AppData\Local\Programs\Microsoft VS Code\Code.exe" "c:\Users\epuxphi\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\json-language-features\server\dist\node\jsonServerMain" --node-ipc --clientProcessId=504
    0	    24	 13272	       c:\Users\epuxphi\AppData\Roaming\Code\User\globalStorage\tabnine.tabnine-vscode\binaries\3.5.5\x86_64-pc-windows-gnu\TabNine.exe --client=vscode --no-lsp=true --client-metadata clientVersion=1.57.1 pluginVersion=3.4.6 t9-vscode-AutoImportEnabled=true t9-vscode-TSAutoImportEnabled=true t9-vscode-JSAutoImportEnabled=true vscode-telemetry-enabled=true vscode-remote=false vscode-remote-name=undefined vscode-extension-kind=1 "vscode-theme-name=Monokai Pro" vscode-theme-kind=Dark vscode-machine-id=7d4fab3af147874b92eb51e5fbd6216226a77ceaa96b61c71985c9c94c96fc47 vscode-is-new-app-install=false vscode-session-id=d205af03-cfa4-4f78-8621-68ab5b495d721624242258023 vscode-language=en "vscode-app-name=Visual Studio Code" vscode-status-customization=unknown ide-restart-counter=0
    0	    12	  7112	         console-window-host (Windows internal process)
    0	    23	  9640	         "C:\Users\epuxphi\AppData\Roaming\Code\User\globalStorage\tabnine.tabnine-vscode\binaries\3.5.5\x86_64-pc-windows-gnu\WD-TabNine.exe" --client vscode --client-metadata clientVersion=1.57.1 pluginVersion=3.4.6 t9-vscode-AutoImportEnabled=true t9-vscode-TSAutoImportEnabled=true t9-vscode-JSAutoImportEnabled=true vscode-telemetry-enabled=true vscode-remote=false vscode-remote-name=undefined vscode-extension-kind=1 "vscode-theme-name=Monokai Pro" vscode-theme-kind=Dark vscode-machine-id=7d4fab3af147874b92eb51e5fbd6216226a77ceaa96b61c71985c9c94c96fc47 vscode-is-new-app-install=false vscode-session-id=d205af03-cfa4-4f78-8621-68ab5b495d721624242258023 vscode-language=en "vscode-app-name=Visual Studio Code" vscode-status-customization=unknown ide-restart-counter=0
    0	   104	 24320	         "c:\Users\epuxphi\AppData\Roaming\Code\User\globalStorage\tabnine.tabnine-vscode\binaries\3.5.5\x86_64-pc-windows-gnu\TabNine.exe" --client=vscode --no-lsp=true --client-metadata clientVersion=1.57.1 pluginVersion=3.4.6 t9-vscode-AutoImportEnabled=true t9-vscode-TSAutoImportEnabled=true t9-vscode-JSAutoImportEnabled=true vscode-telemetry-enabled=true vscode-remote=false vscode-remote-name=undefined vscode-extension-kind=1 "vscode-theme-name=Monokai Pro" vscode-theme-kind=Dark vscode-machine-id=7d4fab3af147874b92eb51e5fbd6216226a77ceaa96b61c71985c9c94c96fc47 vscode-is-new-app-install=false vscode-session-id=d205af03-cfa4-4f78-8621-68ab5b495d721624242258023 vscode-language=en "vscode-app-name=Visual Studio Code" vscode-status-customization=unknown ide-restart-counter=0 --no_bootstrap
    0	   456	  9616	           "c:\Users\epuxphi\AppData\Roaming\Code\User\globalStorage\tabnine.tabnine-vscode\binaries\3.5.5\x86_64-pc-windows-gnu\TabNine-deep-local.exe" --settings C:\Users\epuxphi\AppData\Roaming\TabNine\local.settings --settings-version 1 --client vscode --client-metadata clientVersion=1.57.1 pluginVersion=3.4.6 t9-vscode-AutoImportEnabled=true t9-vscode-TSAutoImportEnabled=true t9-vscode-JSAutoImportEnabled=true vscode-telemetry-enabled=true vscode-remote=false vscode-remote-name=undefined vscode-extension-kind=1 "vscode-theme-name=Monokai Pro" vscode-theme-kind=Dark vscode-machine-id=7d4fab3af147874b92eb51e5fbd6216226a77ceaa96b61c71985c9c94c96fc47 vscode-is-new-app-install=false vscode-session-id=d205af03-cfa4-4f78-8621-68ab5b495d721624242258023 vscode-language=en "vscode-app-name=Visual Studio Code" vscode-status-customization=unknown ide-restart-counter=0
    0	    44	 13028	   utility
    0	    27	 16180	   crashpad-handler
    0	    91	 19876	   shared-process
    0	    65	 19096	     ptyHost
```

</details>
<details>
<summary>Workspace Info</summary>

```
;
```

</details>
<details><summary>Extensions (81)</summary>

Extension|Author (truncated)|Version
---|---|---
rainbow-brackets|2gu|0.0.6
better-comments|aar|2.1.0
html-snippets|abu|0.2.1
codesnap|adp|1.3.2
include-autocomplete|ajs|0.0.4
increment-selection|alb|0.2.0
Bookmarks|ale|13.1.0
auto-add-brackets|ali|0.11.0
All-Autocomplete|Ati|0.0.23
vscode-django|bat|1.6.0
vscode-log-viewer|ber|0.12.2
unique-lines|bib|1.0.0
color-info|bie|0.6.0
xml2json|bui|1.2.4
path-intellisense|chr|2.3.0
highlight-line-vscode|cli|0.0.1
bracket-pair-colorizer|Coe|1.0.61
compulim-vscode-closetag|Com|1.2.0
transformer|dak|1.11.1
python-extension-pack|don|1.6.0
xml|Dot|2.5.1
LogFileHighlighter|emi|2.11.0
vscode-relative-line-numbers|ext|0.0.5
vscode-highlight|fab|1.5.0
file-icons|fil|1.0.29
auto-close-tag|for|0.5.10
auto-complete-tag|for|0.1.0
auto-rename-tag|for|0.1.6
code-runner|for|0.11.4
sort-json-array|fvc|3.0.1
beautify|Hoo|1.5.0
vscode-power-mode|hoo|2.2.0
rest-client|hum|0.24.5
output-colorizer|IBM|0.1.2
highlight-icemode|ice|0.0.6
compareit|in4|0.0.2
path-autocomplete|ion|1.17.1
latex-workshop|Jam|8.19.2
better-cpp-syntax|jef|1.15.6
vscode-insertdatestring|jsy|2.3.0
auto-comment-blocks|kev|1.0.1
vsc-python-indent|Kev|1.14.2
vscode-format-context-menu|lac|1.0.4
vscode-smart-column-indenter|lmc|0.0.13
MagicPython|mag|1.1.0
python-path|mge|0.0.11
HTMLHint|mka|0.10.0
theme-monokai-pro-vscode|mon|1.1.19
python|ms-|2021.6.944021595
vscode-pylance|ms-|2021.6.2
jupyter|ms-|2021.6.999406279
cpptools|ms-|1.4.1
reload|nat|0.0.6
color-highlight|nau|2.3.0
autodocstring|njp|0.5.4
indent-rainbow|ode|7.5.0
subtle-brackets|raf|3.0.0
vscode-commons|red|0.0.6
vscode-xml|red|0.16.1
vscode-sort-json|ric|1.20.0
vscode-statusbar-json-path|ric|1.4.5
datetime|rid|2.0.3
ActiveFileInStatusBar|Ros|1.0.3
partial-diff|ryu|1.4.3
text-marker|ryu|1.11.0
trailing-spaces|sha|0.3.1
indenticator|Sir|0.7.0
vscode-fileutils|sle|3.4.5
vscode-smoothtype|spi|1.2.0
guides|spy|0.9.3
rewrap|stk|1.14.0
tabnine-vscode|Tab|3.4.6
selectline-statusbar|tom|0.0.2
sort-lines|Tyr|1.9.0
errorlens|use|3.2.7
vscodeintellicode|Vis|1.2.14
colonize|vms|2.2.2
vscode-icons|vsc|11.5.0
vim|vsc|1.21.3
vscode-jumpy|wma|0.3.1
clang-format|xav|1.9.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
pythonvspyt700cf:30270857
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt639:30300192
pythontb:30283811
pythonvspyt551cf:30311713
vspre833:30321513
pythonptprofiler:30281270
vshan820:30294714
pythondataviewer:30285071
vscus158:30321503
pythonvsuse255cf:30323309
vscorehov:30309549
vscod805cf:30301675
binariesv615:30323119

```

</details>

**Comment by larskinn** [comment]
Same here, seems like it doesn't respect .editorconfig settings anymore

**Comment by alexdima** [comment]
It looks like the issue is caused by one of your installed extensions. In order to narrow down the possible problematic extension, please try the following:
* run `F1 > Help: Start Extension Bisect`
* (this will restart VS Code with half of your extensions disabled)
* try to do the steps which reproduce the problem
* if the problem reproduces, click "This is bad" (this narrows down the problematic extension to the half which was enabled)
* if the problem does not reproduce, click "Good now" (this narrows down the problematic extension to the half which was disabled)
* these steps will repeat until the problematic extension is found

| initial | step | final |
|---|---|---|
| ![image](https://user-images.githubusercontent.com/5047891/101918658-e9170f80-3bc9-11eb-9977-857baa291937.png) | ![image](https://user-images.githubusercontent.com/5047891/101918683-ef0cf080-3bc9-11eb-9f9e-98b0177991b2.png) | ![image](https://user-images.githubusercontent.com/5047891/101918706-f502d180-3bc9-11eb-8c86-e56355958bd4.png) |

**Comment by volksvagen** [comment]
hello, @alexdima 

it ends here, seems that clang-format is the "crminal':

![image](https://user-images.githubusercontent.com/30553808/122786350-cc7d3200-d2e6-11eb-937e-460760528d48.png)

**Comment by larskinn** [comment]
@alexdima I can reproduce this with no extensions enabled

- Open workspace which has an .editorconfig file at its root (or in one of the workspace's parent directories, that also used to work)
- Command palette, "Developer: Reload with extensions disabled"
- Create a new file with a file type specified in .editorconfig.
   - In my case: `SomeComponent.tsx` 
- Observe "Tab size: 8" in the bottom right corner
   - In my case: should be "Spaces: 4"

**Comment by alexdima** [comment]
@volksvagen Please report directly against the extension

@larskinn VS Code does not honor `.editorconfig` files in any way. It is the editorconfig extension that reads those files and then uses vscode APIs to change the indentation settings of the current opened file. Can you check your user settings, folder settings and workspace settings for `editor.tabSize`, `editor.insertSpaces` and `editor.detectIndentation` ?

**Comment by larskinn** [comment]
> VS Code does not honor `.editorconfig` files in any way [...]

@alexdima Okay, sorry for the wild goose chase. I actually don't have the editorconfig extension, but somehow the default for new files, which used to be "Spaces: 4", has changed to "Tab size: 8" with the latest update. Not a huge problem for me, as it is fixed by saving (autoformats with prettier, which reads `.editorconfig`) and closing/reopening the file (VS Code autodetects the correct indentation)

I checked my user settings:
- `editor.tabSize`
  - was changed to 8, from the default 4
- `editor.insertSpaces`
  - was changed to `false`, from the default `true`
- `editor.detectIndentation`
  - was unchanged from default `true` 

I can honestly say that I have never changed these settings myself, and the default indentation for new files until the latest VS Code update was always 4 spaces. So either VS Code or some extension somehow changed my user settings immediately after the latest update, or these settings were not being respected before.

After resetting the above settings to default, it works as expected.
